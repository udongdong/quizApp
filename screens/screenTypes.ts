import {NavigatorScreenParams} from '@react-navigation/native';

export type MainStackParamList = {
  quizScreen: undefined;
  mainTab: NavigatorScreenParams<MainTabParamList>;
};

export type MainTabParamList = {
  quiz: undefined;
  history: undefined;
  review: undefined;
};
