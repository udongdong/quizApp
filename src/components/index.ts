export * from './Button';
export * from './ProgressBar';
export * from './RadioButton';
export * from './Space';
export * from './quiz';
