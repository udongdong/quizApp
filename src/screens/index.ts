export * from './MainScreen';
export * from './MainTab';
export * from './quiz';
export * from './review';
export * from './screenTypes';
