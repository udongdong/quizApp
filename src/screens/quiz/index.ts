export * from './QuizMain';
export * from './QuizResultScreen';
export * from './quizScreen';
