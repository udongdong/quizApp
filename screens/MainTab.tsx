import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import React from 'react';
import {MainTabParamList} from './screenTypes';
import QuizMain from './quiz/QuizMain';

export default function MainTab(): React.JSX.Element {
  const Tab = createBottomTabNavigator<MainTabParamList>();

  return (
    <Tab.Navigator>
      <Tab.Screen name="quiz" component={QuizMain} />
    </Tab.Navigator>
  );
}
