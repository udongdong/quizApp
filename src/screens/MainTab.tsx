import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import React from 'react';
import {MainTabParamList} from './screenTypes';
import QuizMain from './quiz/QuizMain';
import ReviewMain from './review/ReviewMain';
import {color} from '../styles';

export default function MainTab(): React.JSX.Element {
  const Tab = createBottomTabNavigator<MainTabParamList>();

  return (
    <Tab.Navigator
      screenOptions={{
        tabBarLabelPosition: 'beside-icon',
        tabBarLabelStyle: {
          fontWeight: '700',
          fontSize: 15,
        },
        tabBarIconStyle: {display: 'none'},
        tabBarStyle: {height: 60},
        tabBarActiveTintColor: color.primary,
        headerTitleStyle: {
          color: color.primary,
        },
      }}>
      <Tab.Screen
        name="quiz"
        component={QuizMain}
        options={{tabBarLabel: '퀴즈', title: '퀴즈 풀기'}}
      />
      <Tab.Screen
        name="review"
        component={ReviewMain}
        options={{tabBarLabel: '리뷰', title: '퀴즈 리뷰'}}
      />
    </Tab.Navigator>
  );
}
