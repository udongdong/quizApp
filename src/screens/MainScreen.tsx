import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {MainStackParamList} from './screenTypes';
import {MainTab, ReviewDetail, QuizScreen} from './';
import {QuizResultScreen} from './quiz';

export function MainScreen(): React.JSX.Element {
  const MainStack = createNativeStackNavigator<MainStackParamList>();

  return (
    <NavigationContainer>
      <MainStack.Navigator
        initialRouteName="mainTab"
        screenOptions={{headerShown: false}}>
        <MainStack.Screen name="mainTab" component={MainTab} />
        <MainStack.Screen name="quizScreen" component={QuizScreen} />
        <MainStack.Screen
          name="quizResultScreen"
          component={QuizResultScreen}
        />
        <MainStack.Screen name="reviewDetail" component={ReviewDetail} />
      </MainStack.Navigator>
    </NavigationContainer>
  );
}
