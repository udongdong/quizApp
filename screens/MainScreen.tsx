import {createNativeStackNavigator} from '@react-navigation/native-stack';
import React from 'react';
import {MainStackParamList} from './screenTypes';
import {NavigationContainer} from '@react-navigation/native';
import MainTab from './MainTab';
import QuizScreen from './quiz/QuizScreen';
import {QuizResultScreen} from './quiz/QuizResultScreen';
import ReviewDetail from './review/ReviewDetail';

export default function MainScreen(): React.JSX.Element {
  const MainStack = createNativeStackNavigator<MainStackParamList>();

  return (
    <NavigationContainer>
      <MainStack.Navigator
        initialRouteName="mainTab"
        screenOptions={{headerShown: false}}>
        <MainStack.Screen name="mainTab" component={MainTab} />
        <MainStack.Screen name="quizScreen" component={QuizScreen} />
        <MainStack.Screen
          name="quizResultScreen"
          component={QuizResultScreen}
        />
        <MainStack.Screen name="reviewDetail" component={ReviewDetail} />
      </MainStack.Navigator>
    </NavigationContainer>
  );
}
