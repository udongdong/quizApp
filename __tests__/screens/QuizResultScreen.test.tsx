import 'react-native';
import React from 'react';
import {QuizResultScreen} from '../../src/screens';

import {render} from '@testing-library/react-native';
import {MockStackNavigator} from './MockNavigator';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {MainStackParamList} from 'screens/screenTypes';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useRoute: () => ({
    params: {
      correct: 1,
      total: 10,
      time: 145,
    },
  }),
}));

function getComponent() {
  const MainStack = createNativeStackNavigator<MainStackParamList>();

  return (
    <MockStackNavigator>
      <MainStack.Screen name="quizResultScreen" component={QuizResultScreen} />
    </MockStackNavigator>
  );
}

describe('QuizResultScreen test', () => {
  const component = getComponent();

  test('render QuizResultScreen', () => {
    const rendered = render(component);
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});
