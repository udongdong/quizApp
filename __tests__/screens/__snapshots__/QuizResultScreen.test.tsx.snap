// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizResultScreen test render QuizResultScreen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 40,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              퀴즈가 끝났습니다. 
수고하셨습니다
            </Text>
            <View
              style={
                {
                  "marginLeft": 0,
                  "marginTop": 30,
                }
              }
            />
            <Text
              style={
                {
                  "fontSize": 20,
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              정답 수 : 1
               /
               
              총 문항수 : 10
            </Text>
            <Text
              style={
                {
                  "fontSize": 20,
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              소요 시간: 02:25
            </Text>
            <View
              style={
                {
                  "alignSelf": "center",
                  "bottom": 0,
                  "height": 70,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#5F37FF",
                    "borderWidth": 0,
                    "bottom": 0,
                    "flexDirection": "row",
                    "height": 70,
                    "justifyContent": "center",
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  홈 으로 이동
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="quizResultScreen"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
